Makefile.common

```diff
@@ -16,36 +16,41 @@ endif
 # Makefile.config
 -include $(HACL_HOME)/Makefile.config
 
+# Essentially FSTAR_EXE ?= fstar.exe, but we want to
+# 1) Make it absolute
+# 2) Fail early if not found
+ifndef FSTAR_EXE
+FSTAR_EXE := $(shell which fstar.exe)
+ifneq ($(.SHELLSTATUS),0)
+  $(error "Did not find F* in PATH, add it or set FSTAR_EXE")
+endif
+endif
+
 ifndef KRML_HOME
 # assuming an Everest source tree
 KRML_HOME 	:= $(HACL_HOME)/../karamel
 endif
 
-ifndef FSTAR_HOME
-  FSTAR_EXE=$(shell which fstar.exe)
-  ifeq ($(FSTAR_EXE),)
-    # F* not found in PATH, assuming an Everest source tree
-    FSTAR_HOME := $(HACL_HOME)/../FStar
-  else
-    # F* found in PATH, assuming directory name ends with /bin
-    FSTAR_HOME := $(dir $(FSTAR_EXE))/..
-  endif
-endif
-
-FSTAR_EXE := $(FSTAR_HOME)/bin/fstar.exe
-
 # Check that the relevant homes and executables at least exist.
 mustexist =						\
   $(if $(wildcard $(value $(strip $1))),,		\
     $(error $1 ($(value $(strip $1))) does not exist))
 
-$(call mustexist, FSTAR_HOME)
+# eval $(FSTAR_OCAMLENV) extends the environment so we can
+# find fstar's ocaml libraries.
+FSTAR_OCAMLENV := $(shell $(FSTAR_EXE) --ocamlenv)
+
+# We also set this variable to reduce warnings
+FSTAR_OCAMLENV += OCAMLFIND_IGNORE_DUPS_IN='$(shell ocamlc -where)/compiler-libs';
+FSTAR_OCAMLENV += export OCAMLFIND_IGNORE_DUPS_IN;
```

Comment:
how about preserving the previous value of OCAMLFIND_IGNORE_DUPS_IN, if any? (I recently started seeing errors about something being both in lib/ and compiler-libs/ and I'm wondering if this might be it)

Reply:
Pushed a patch, fwiw it was also overriding the original value before (but the variable was unset anyway, for me). I've also seen those warnings for a long time I think.